src/luascript.cpp

```diff
@@ -11683,13 +11683,13 @@ int LuaScriptInterface::luaConditionAddDamage(lua_State* L)
 int LuaScriptInterface::luaMonsterTypeCreate(lua_State* L)
 {
 	// MonsterType(id or name)
-	MonsterType* monsterType;
-	if (isNumber(L, 2)) {
-		monsterType = g_monsters.getMonsterType(getNumber<uint32_t>(L, 2));
-	} else {
-		monsterType = g_monsters.getMonsterType(getString(L, 2));
+	
+	if (!isString(L, 2)) {
+		lua_pushnil(L);
+		return 1;
 	}
-
+	auto monsterType = g_monsters.getMonsterType(getString(L, 2));
```

Comment:
Well, the type is already "encoded" in the name of the variable and the return value of the function (and the function name itself), I though it would be needless to say it even once more.